federation/pkg/kubefed/init/init.go

```diff
@@ -576,6 +586,43 @@ func createControllerManager(clientset *client.Clientset, namespace, name, svcNa
 	return clientset.Extensions().Deployments(namespace).Create(dep)
 }
 
+func waitForPods(clientset *client.Clientset, fedPods []string, namespace string) error {
+	err := wait.PollInfinite(podWaitInterval, func() (bool, error) {
+		podList, err := clientset.Core().Pods(namespace).List(api.ListOptions{})
+		if err != nil {
+			return false, nil
+		}
+		for _, pod := range podList.Items {
+			for _, fedPod := range fedPods {
+				if strings.HasPrefix(pod.Name, fedPod) && pod.Status.Phase == "Running" {
+					return true, nil
```

Comment:
As I was pointing out in my previous comment, you will need additional logic here. What is happening here is, the poll loop exits due to the return statement as soon as a single `Running` pod is found. However, you need to verify that both the pods are running.